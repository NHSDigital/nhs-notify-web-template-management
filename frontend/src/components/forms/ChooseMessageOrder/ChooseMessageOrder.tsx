'use client';

import { useActionState, useState } from 'react';
import { NHSNotifyRadioButtonForm } from '@molecules/NHSNotifyRadioButtonForm/NHSNotifyRadioButtonForm';
import { NhsNotifyErrorSummary } from '@molecules/NhsNotifyErrorSummary/NhsNotifyErrorSummary';
import content from '@content/content';
import { ErrorState } from 'nhs-notify-web-template-management-utils';
import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';
import {
  $ChooseMessageOrder,
  chooseMessageOrderAction,
  MessageOrder,
  MESSAGE_ORDER_OPTIONS_LIST,
} from './server-action';
import { validate } from '@utils/client-validate-form';

export const messageOrderDisplayMappings: Record<MessageOrder, string> = {
  NHSAPP: 'NHS App only',
  'NHSAPP,EMAIL': 'NHS App, Email',
  'NHSAPP,SMS': 'NHS App, Text message',
  'NHSAPP,EMAIL,SMS': 'NHS App, Email, Text message',
  'NHSAPP,SMS,EMAIL': 'NHS App, Text message, Email',
  'NHSAPP,SMS,LETTER': 'NHS App, Text message, Letter',
  'NHSAPP,EMAIL,SMS,LETTER': 'NHS App, Email, Text message, Letter',
  LETTER: 'Letter only',
};

export const ChooseMessageOrder = () => {
  const [state, action] = useActionState(chooseMessageOrderAction, {});
  const [errorState, setErrorState] = useState<ErrorState | undefined>(
    state.errorState
  );

  const formValidate = validate($ChooseMessageOrder, setErrorState);

  const options = MESSAGE_ORDER_OPTIONS_LIST.map((messageOrder) => ({
    id: messageOrder,
    text: messageOrderDisplayMappings[messageOrder],
  }));

  const { pageHeading, buttonText, hint, backLinkText } =
    content.components.chooseMessageOrder;

  return (
    <NHSNotifyMain>
      <NhsNotifyErrorSummary errorState={errorState} />
      <NHSNotifyRadioButtonForm
        formId='choose-message-order'
        radiosId='messageOrder'
        action={action}
        state={{ errorState }}
        pageHeading={pageHeading}
        options={options}
        buttonText={buttonText}
        hint={hint}
        formAttributes={{ onSubmit: formValidate }}
        backLink={{
          text: backLinkText,
          url: '/message-plans',
        }}
      />
    </NHSNotifyMain>
  );
};
