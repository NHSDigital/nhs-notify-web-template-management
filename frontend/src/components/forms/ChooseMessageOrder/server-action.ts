import { redirect, RedirectType } from 'next/navigation';
import { FormState } from 'nhs-notify-web-template-management-utils';
import { z } from 'zod';
import content from '@content/content';

export const MESSAGE_ORDER_OPTIONS_LIST = [
  'NHS_APP',
  'NHS_APP,EMAIL',
  'NHS_APP,SMS',
  'NHS_APP,EMAIL,SMS',
  'NHS_APP,SMS,EMAIL',
  'NHS_APP,SMS,LETTER',
  'NHS_APP,EMAIL,SMS,LETTER',
  'LETTER',
] as const;

export type MessageOrder = (typeof MESSAGE_ORDER_OPTIONS_LIST)[number];

export const $ChooseMessageOrder = z.object({
  messageOrder: z.enum(MESSAGE_ORDER_OPTIONS_LIST, {
    message: content.components.chooseMessageOrder.form.messageOrder.error,
  }),
});

export async function chooseMessageOrderAction(
  _: FormState,
  formData: FormData
): Promise<FormState> {
  const parsedForm = $ChooseMessageOrder.safeParse(
    Object.fromEntries(formData.entries())
  );

  if (!parsedForm.success) {
    return {
      errorState: z.flattenError(parsedForm.error),
    };
  }

  redirect(
    `/message-plans/create-message-plan?messageOrder=${encodeURIComponent(parsedForm.data.messageOrder)}`,
    RedirectType.push
  );
}
