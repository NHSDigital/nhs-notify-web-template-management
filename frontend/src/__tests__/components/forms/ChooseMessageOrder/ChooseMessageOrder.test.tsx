import { useActionState } from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ChooseMessageOrder } from '@forms/ChooseMessageOrder/ChooseMessageOrder';
import { TemplateFormState } from 'nhs-notify-web-template-management-utils';

jest.mock('@utils/amplify-utils');

jest.mock('react', () => {
  const originalModule = jest.requireActual('react');

  return {
    ...originalModule,
    useActionState: jest
      .fn()
      .mockImplementation(
        (
          _: (
            formState: TemplateFormState,
            formData: FormData
          ) => Promise<TemplateFormState>,
          initialState: TemplateFormState
        ) => [initialState, '/action']
      ),
  };
});

describe('Choose message order page', () => {
  it('renders form', () => {
    const container = render(<ChooseMessageOrder />);
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders error component', () => {
    const mockUseActionState = jest.fn().mockReturnValue([
      {
        errorState: {
          formErrors: [],
          fieldErrors: {
            page: ['Component error message'],
          },
        },
      },
      '/action',
    ]);

    jest.mocked(useActionState).mockImplementation(mockUseActionState);

    const container = render(<ChooseMessageOrder />);
    expect(container.asFragment()).toMatchSnapshot();
  });

  test('Client-side validation triggers', () => {
    const container = render(<ChooseMessageOrder />);
    const submitButton = screen.getByTestId('submit-button');
    fireEvent.click(submitButton);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
