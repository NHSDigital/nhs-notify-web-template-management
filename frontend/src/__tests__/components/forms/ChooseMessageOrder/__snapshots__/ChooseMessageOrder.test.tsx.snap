// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose message order page Client-side validation triggers 1`] = `
<DocumentFragment>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      aria-labelledby="error-summary-title"
      class="nhsuk-error-summary"
      role="alert"
      tabindex="-1"
    >
      <h2
        class="nhsuk-error-summary__title"
        data-testid="error-summary"
        id="error-summary-title"
      >
        There is a problem
      </h2>
      <ul
        class="nhsuk-list nhsuk-error-summary__list"
      >
        <li>
          <a
            href="#messageOrder"
          >
            Select a message order
          </a>
        </li>
      </ul>
    </div>
    <form
      action="/action"
    >
      <input
        name="form-id"
        readonly=""
        type="hidden"
        value="choose-message-order"
      />
      <input
        name="csrf_token"
        readonly=""
        type="hidden"
        value="no_token"
      />
      <div
        class="nhsuk-form-group nhsuk-form-group--error"
      >
        <fieldset
          class="nhsuk-fieldset"
        >
          <legend
            class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
            data-testid="messageOrder-form__legend"
          >
            <h1
              class="nhsuk-fieldset__heading"
            >
              Choose a message order
            </h1>
          </legend>
          <div
            class="nhsuk-form-group nhsuk-form-group--error"
          >
            <div
              class="nhsuk-hint"
              id="messageOrder--hint"
            >
              Select one option
            </div>
            <span
              class="nhsuk-error-message"
              id="messageOrder--error-message"
            >
              <span
                class="nhsuk-u-visually-hidden"
              >
                Error: 
              </span>
              Select a message order
            </span>
            <div
              aria-describedby="messageOrder--hint messageOrder--error-message"
              class="nhsuk-radios"
              id="messageOrder"
            >
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP-radio"
                  id="messageOrder-NHSAPP"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP"
                  id="messageOrder-NHSAPP--label"
                >
                  NHS App only
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL-radio"
                  id="messageOrder-NHSAPP,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL"
                  id="messageOrder-NHSAPP,EMAIL--label"
                >
                  NHS App, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS-radio"
                  id="messageOrder-NHSAPP,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS"
                  id="messageOrder-NHSAPP,SMS--label"
                >
                  NHS App, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS"
                  id="messageOrder-NHSAPP,EMAIL,SMS--label"
                >
                  NHS App, Email, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,EMAIL-radio"
                  id="messageOrder-NHSAPP,SMS,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,EMAIL"
                  id="messageOrder-NHSAPP,SMS,EMAIL--label"
                >
                  NHS App, Text message, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,LETTER"
                  id="messageOrder-NHSAPP,SMS,LETTER--label"
                >
                  NHS App, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER--label"
                >
                  NHS App, Email, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="LETTER-radio"
                  id="messageOrder-LETTER"
                  name="messageOrder"
                  type="radio"
                  value="LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-LETTER"
                  id="messageOrder-LETTER--label"
                >
                  Letter only
                </label>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        aria-disabled="false"
        class="nhsuk-button"
        data-testid="submit-button"
        id="choose-message-order-submit-button"
        type="submit"
      >
        Save and continue
      </button>
      <a
        class="inline-block nhsuk-u-font-size-19 nhsuk-u-margin-left-3 nhsuk-u-padding-top-3"
        href="/message-plans"
      >
        Go back
      </a>
    </form>
  </main>
</DocumentFragment>
`;

exports[`Choose message order page renders error component 1`] = `
<DocumentFragment>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      aria-labelledby="error-summary-title"
      class="nhsuk-error-summary"
      role="alert"
      tabindex="-1"
    >
      <h2
        class="nhsuk-error-summary__title"
        data-testid="error-summary"
        id="error-summary-title"
      >
        There is a problem
      </h2>
      <ul
        class="nhsuk-list nhsuk-error-summary__list"
      >
        <li>
          <a
            href="#page"
          >
            Component error message
          </a>
        </li>
      </ul>
    </div>
    <form
      action="/action"
    >
      <input
        name="form-id"
        readonly=""
        type="hidden"
        value="choose-message-order"
      />
      <input
        name="csrf_token"
        readonly=""
        type="hidden"
        value="no_token"
      />
      <div
        class="nhsuk-form-group"
      >
        <fieldset
          class="nhsuk-fieldset"
        >
          <legend
            class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
            data-testid="messageOrder-form__legend"
          >
            <h1
              class="nhsuk-fieldset__heading"
            >
              Choose a message order
            </h1>
          </legend>
          <div
            class="nhsuk-form-group"
          >
            <div
              class="nhsuk-hint"
              id="messageOrder--hint"
            >
              Select one option
            </div>
            <div
              aria-describedby="messageOrder--hint"
              class="nhsuk-radios"
              id="messageOrder"
            >
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP-radio"
                  id="messageOrder-NHSAPP"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP"
                  id="messageOrder-NHSAPP--label"
                >
                  NHS App only
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL-radio"
                  id="messageOrder-NHSAPP,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL"
                  id="messageOrder-NHSAPP,EMAIL--label"
                >
                  NHS App, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS-radio"
                  id="messageOrder-NHSAPP,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS"
                  id="messageOrder-NHSAPP,SMS--label"
                >
                  NHS App, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS"
                  id="messageOrder-NHSAPP,EMAIL,SMS--label"
                >
                  NHS App, Email, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,EMAIL-radio"
                  id="messageOrder-NHSAPP,SMS,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,EMAIL"
                  id="messageOrder-NHSAPP,SMS,EMAIL--label"
                >
                  NHS App, Text message, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,LETTER"
                  id="messageOrder-NHSAPP,SMS,LETTER--label"
                >
                  NHS App, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER--label"
                >
                  NHS App, Email, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="LETTER-radio"
                  id="messageOrder-LETTER"
                  name="messageOrder"
                  type="radio"
                  value="LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-LETTER"
                  id="messageOrder-LETTER--label"
                >
                  Letter only
                </label>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        aria-disabled="false"
        class="nhsuk-button"
        data-testid="submit-button"
        id="choose-message-order-submit-button"
        type="submit"
      >
        Save and continue
      </button>
      <a
        class="inline-block nhsuk-u-font-size-19 nhsuk-u-margin-left-3 nhsuk-u-padding-top-3"
        href="/message-plans"
      >
        Go back
      </a>
    </form>
  </main>
</DocumentFragment>
`;

exports[`Choose message order page renders form 1`] = `
<DocumentFragment>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <form
      action="/action"
    >
      <input
        name="form-id"
        readonly=""
        type="hidden"
        value="choose-message-order"
      />
      <input
        name="csrf_token"
        readonly=""
        type="hidden"
        value="no_token"
      />
      <div
        class="nhsuk-form-group"
      >
        <fieldset
          class="nhsuk-fieldset"
        >
          <legend
            class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
            data-testid="messageOrder-form__legend"
          >
            <h1
              class="nhsuk-fieldset__heading"
            >
              Choose a message order
            </h1>
          </legend>
          <div
            class="nhsuk-form-group"
          >
            <div
              class="nhsuk-hint"
              id="messageOrder--hint"
            >
              Select one option
            </div>
            <div
              aria-describedby="messageOrder--hint"
              class="nhsuk-radios"
              id="messageOrder"
            >
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP-radio"
                  id="messageOrder-NHSAPP"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP"
                  id="messageOrder-NHSAPP--label"
                >
                  NHS App only
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL-radio"
                  id="messageOrder-NHSAPP,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL"
                  id="messageOrder-NHSAPP,EMAIL--label"
                >
                  NHS App, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS-radio"
                  id="messageOrder-NHSAPP,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS"
                  id="messageOrder-NHSAPP,SMS--label"
                >
                  NHS App, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS"
                  id="messageOrder-NHSAPP,EMAIL,SMS--label"
                >
                  NHS App, Email, Text message
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,EMAIL-radio"
                  id="messageOrder-NHSAPP,SMS,EMAIL"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,EMAIL"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,EMAIL"
                  id="messageOrder-NHSAPP,SMS,EMAIL--label"
                >
                  NHS App, Text message, Email
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,SMS,LETTER"
                  id="messageOrder-NHSAPP,SMS,LETTER--label"
                >
                  NHS App, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="NHSAPP,EMAIL,SMS,LETTER-radio"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  name="messageOrder"
                  type="radio"
                  value="NHSAPP,EMAIL,SMS,LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-NHSAPP,EMAIL,SMS,LETTER"
                  id="messageOrder-NHSAPP,EMAIL,SMS,LETTER--label"
                >
                  NHS App, Email, Text message, Letter
                </label>
              </div>
              <div
                class="nhsuk-radios__item"
              >
                <input
                  class="nhsuk-radios__input"
                  data-testid="LETTER-radio"
                  id="messageOrder-LETTER"
                  name="messageOrder"
                  type="radio"
                  value="LETTER"
                />
                <label
                  class="nhsuk-label nhsuk-radios__label"
                  for="messageOrder-LETTER"
                  id="messageOrder-LETTER--label"
                >
                  Letter only
                </label>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        aria-disabled="false"
        class="nhsuk-button"
        data-testid="submit-button"
        id="choose-message-order-submit-button"
        type="submit"
      >
        Save and continue
      </button>
      <a
        class="inline-block nhsuk-u-font-size-19 nhsuk-u-margin-left-3 nhsuk-u-padding-top-3"
        href="/message-plans"
      >
        Go back
      </a>
    </form>
  </main>
</DocumentFragment>
`;
